src/Microsoft.Data.SqlClient/tests/ManualTests/makepfxcert.ps1

```diff
@@ -0,0 +1,140 @@
+﻿# Licensed to the .NET Foundation under one or more agreements.
+# The .NET Foundation licenses this file to you under the MIT license.
+# See the LICENSE file in the project root for more information.
+# Script: Invoke-SqlServerCertificateCommand# 
+# Author: SqlClient Team
+# Date: March 20, 2024
+# Comments: This scripts creates SSL Self-Signed Certificate for TestTdsServer in pfx format.
+# This script is not intended to be used in any production environments.
+
+param ($OutDir)
+
+function Invoke-SqlServerCertificateCommand {
+  [CmdletBinding()]
+  param(
+    [Parameter(Mandatory = $false)]
+    [string] $certificateName = "localhostcert.cer",
+    [string] $myCertStoreLocation = "Cert:\LocalMachine\My",
+    [string] $rootCertStoreLocation = "Cert:\LocalMachine\Root",
+    [string] $sqlAliasName = "SQLAliasName",
+    [string] $localhost = "localhost",
+    [string] $LoopBackIPV4 = "127.0.0.1",
+    [string] $LoopBackIPV6 = "::1"
+  )
+  Write-Output "Certificate generation started..."
+
+  # Change directory to where the tests are
+  Write-Output "Change directory to $OutDir ..."
+  cd $OutDir
+  pwd
+
+  try {
+    # Get FQDN of the machine
+    Write-Output "Get FQDN of the machine..."
+    $fqdn = [System.Net.Dns]::GetHostByName(($env:computerName)).HostName
+    Write-Output "FQDN = $fqdn"
+
+    $OS = [System.Environment]::OSVersion.Platform
+    Write-Output "Operating System is $OS"
+
+    # Create a self-signed certificate
+    if ($OS -eq "Unix") {
+        # Install OpenSSL module
+        Install-Module -Name OpenSSL -Repository PSGallery -Force
+        # Show version of OpenSSL just to make sure it is installed
+        openssl version
+
+        # Create self signed certificate using openssl
+        Write-Output "Creating certificate for linux..."
+        if ($fqdn.length -gt 64) {
+            $machineId = $fqdn.Substring(0,15)
+            openssl req -x509 -newkey rsa:4096 -sha256 -days 1095 -nodes -keyout $OutDir/localhostcert.key -out $OutDir/localhostcert.cer -subj "/CN=$machineId" -addext "subjectAltName=DNS:$fqdn,DNS:localhost,IP:127.0.0.1,IP:::1"
+        }
+        else {
+            openssl req -x509 -newkey rsa:4096 -sha256 -days 1095 -nodes -keyout $OutDir/localhostcert.key -out $OutDir/localhostcert.cer -subj "/CN=$fqdn" -addext "subjectAltName=DNS:$fqdn,DNS:localhost,IP:127.0.0.1,IP:::1"
+        }
+        # Export the certificate to pfx
+        Write-Output "Exporting certificate to pfx..."
+        openssl pkcs12 -export -in $OutDir/localhostcert.cer -inkey $OutDir/localhostcert.key -out $OutDir/localhostcert.pfx -password pass:nopassword
+
+        Write-Output "Converting certificate to pem..."
+        # Create pem from cer
+        cp $OutDir/localhostcert.cer $OutDir/localhostcert.pem
+
+        # Add trust to the pem certificate
+        Write-Output "Adding trust to pem certificate..."
+        openssl x509 -trustout -addtrust "serverAuth" -in $OutDir/localhostcert.pem
```

Comment:
I think you need to add trust to certificate before copying that to root folder.